src/arith_uint256.cpp

```diff
@@ -148,13 +149,18 @@ double base_uint<BITS>::getdouble() const
 template <unsigned int BITS>
 std::string base_uint<BITS>::GetHex() const
 {
-    return ArithToUint256(*this).GetHex();
+    uint256 b;
+    for(int x=0; x<WIDTH; ++x)
+        WriteLE32(b.begin() + x*4, pn[x]);
+    return b.GetHex();
 }
 
 template <unsigned int BITS>
 void base_uint<BITS>::SetHex(const char* psz)
 {
-    *this = UintToArith256(uint256S(psz));
+    uint256 a(uint256S(psz));
+    for(int x=0; x<WIDTH; ++x)
+        pn[x] = ReadLE32(a.begin() + x*4);
```

Comment:
Here it's wrong, it should copy the lowest not he highest bits, i would prefer similar to `uint256::SetHex` here. Probably it should get bytes count.